### ISSUE TITLE
Swerve Module

### ISSUE THREAD

**Issue by NoahSimon8** [Team-Optix-3749/Robot-Code-Template#1]
update this and it's IO to better much the example one. Also do any necessary cleaning

- Complete

**Comment by NoahSimon8** [comment]
Should the sparkmax turning encoder be used? Is it better than the CANCoder in terms up refreshrate and latency?

**Comment by NoahSimon8** [comment]
Is max motor volts actually 12? What happens if we go past this? Can they handle, say, 14 volts and a 50 amp current limit? We'd go pretty fast if that actually just works

**Comment by SlushEE0** [comment]
for the voltage part it should be completely possible. As long as we don't exceed the breakdown voltage of the coil/inductors and limit the current appropriately (higher voltage = more current draw ... and we already saw brownouts with a couple amps over our current current limit. This can be tuned I believe).

**Comment by NoahSimon8** [comment]
> for the voltage part it should be completely possible. As long as we don't exceed the breakdown voltage of the coil/inductors and limit the current appropriately (higher voltage = more current draw ... and we already saw brownouts with a couple amps over our current current limit. This can be tuned I believe).

Ya, i forgot but i actually kinda asked a similar thing on CD a while ago that got some pretty good answers: https://www.chiefdelphi.com/t/maximum-recomended-voltage-for-neo-motors/451165

Let's mess around over break.

**Comment by SlushEE0** [comment]
Oooh variable current limits might be really good. Tanay was also asking for some kind of turbo mode that could cut power to everything but swerve for some extra speed

**Comment by NoahSimon8** [comment]
Nah, i dont think that's a thing because we already do that--motors are set to 0 volts. The only thing drawing power at that point is LED and cams, which relatively speaking are insignificant. Maybe im completely missing some concept but pretty sure its ight.

The current limits could be sick tho. Could def up it to 50/60 for auto, and then maybe have it based on temp and match time after that
